src/main/java/teammates/ui/newcontroller/GetStudentProfileAction.java

```diff
@@ -0,0 +1,48 @@
+package teammates.ui.newcontroller;
+
+import teammates.common.datatransfer.attributes.StudentProfileAttributes;
+import teammates.common.util.Const;
+
+/**
+ * Action: Get a student's profile.
+ */
+public class GetStudentProfileAction extends Action {
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Anyone logged in can get a student's profile
+    }
+
+    @Override
+    public ActionResult execute() {
+        String studentId = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_ID);
```

Comment:
This mean anyone can get anyone else's student profile? Should have a check if it is the same as `userInfo`'s ID.